rr_iarrc/src/finish_line_watcher/finish_line_watcher.cpp

```diff
@@ -78,30 +73,118 @@ void ImageCB(const sensor_msgs::ImageConstPtr& msg) {
         state = HIGH;
-    } else if (state == HIGH && count < 2000) {
+    } else if (state == HIGH && !detected) {
+        incrementCrossNum = !publish_when_detected;
         state = LOW;
+    }
+
+    if (incrementCrossNum) {
         number_of_crosses++;
+        cooldown = cooldown_value;
         ROS_INFO_STREAM("Finish line crossed: " << to_string(number_of_crosses));
     }
 
-    // if(count > 2000 && width > 600) {
-    //     number_of_crosses = 1;
-    //     ROS_INFO_STREAM("Finish line crossed: " <<
-    //     to_string(number_of_crosses));
-    // }
+    if (cooldown > 0)
+        cooldown--;
+
+    // Publish stuff
+    std_msgs::Int8 intmsg;
+    intmsg.data = number_of_crosses;
+
+    if (debug_pub.getNumSubscribers() > 0) {
+        // Draw some debug info
+        if (drawRectDebug) {
+            std::string angle_str = std::string("Angle: ") + std::to_string(angle);
+            cv::putText(debugDrawing, angle_str, cv::Point(5, 100), cv::FONT_HERSHEY_SIMPLEX, 2,
+                        cv::Scalar(0, 143, 143), 2);
+            std::string width_str = std::string("Width: ") + std::to_string(size.width);
+            cv::putText(debugDrawing, width_str, cv::Point(5, 200), cv::FONT_HERSHEY_SIMPLEX, 2,
+                        cv::Scalar(0, 143, 143), 2);
+            std::string height_str = std::string("Height: ") + std::to_string(size.height);
+            cv::putText(debugDrawing, height_str, cv::Point(5, 300), cv::FONT_HERSHEY_SIMPLEX, 2,
+                        cv::Scalar(0, 143, 143), 2);
+        }
+        std::string mode_str =
+              std::string("Mode: ") + std::string(publish_when_detected ? "when detected" : "when crossed");
+        cv::putText(debugDrawing, mode_str, cv::Point(5, 400), cv::FONT_HERSHEY_SIMPLEX, 2, cv::Scalar(0, 143, 143), 2);
+        std::string detected_str = std::string("Detected: ") + std::to_string(detected);
+        cv::putText(debugDrawing, detected_str, cv::Point(5, 500), cv::FONT_HERSHEY_SIMPLEX, 2,
+                    (detected ? cv::Scalar(0, 255, 0) : cv::Scalar(0, 143, 143)), 2);
+        std::string count_str = std::string("Num detections: ") + std::to_string(number_of_crosses);
+        cv::putText(debugDrawing, count_str, cv::Point(5, 600), cv::FONT_HERSHEY_SIMPLEX, 2, cv::Scalar(0, 143, 143),
+                    2);
+
+        // Convert to ros image format and publish
+        debug_img.header = msg->header;
+        debug_img.encoding = "bgr8";
+        debug_img.image = debugDrawing;
+        debug_pub.publish(debug_img.toImageMsg());
+    }
+
+    if (crosses_pub.getNumSubscribers() > 0)
+        crosses_pub.publish(intmsg);
 }
 
 int main(int argc, char** argv) {
-    init(argc, argv, "finish_line_watcher");
+    ros::init(argc, argv, "finish_line_watcher");
 
-    NodeHandle nh;
-    NodeHandle nhp("~");
+    ros::NodeHandle nh;
+    ros::NodeHandle nhp("~");
 
     string img_topic;
     nhp.getParam("img_topic", img_topic);
```

Comment:
I think it would be nice to use the `assertions` helper functions to make sure that these parameters are set

Reply:
Honestly, the only places `assertions` are really used in our codebase is in Evan's code and we should probably use them more, but I wasn't going to strictly push it in this PR. I will leave it up to Dylan's discretion.